provisioning/service/src/Authentication/ServiceAuthenticationWithSharedAccessPolicyToken.cs

```diff
@@ -38,10 +38,7 @@ public string Token
 
         public ServiceConnectionStringBuilder Populate(ServiceConnectionStringBuilder provisioningConnectionStringBuilder)
         {
-            if (provisioningConnectionStringBuilder == null)
-            {
-                throw new ArgumentNullException(nameof(provisioningConnectionStringBuilder));
-            }
+            Argument.AssertNotNull(provisioningConnectionStringBuilder, nameof(provisioningConnectionStringBuilder));
```

Comment:
Same with this, unfortunately. Internal class, so not a public API. Where this might bubble up to a customer will make no sense to them.

I know you didn't change the behavior, just using a library we have, but it isn't the right long-term change we need.